pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotBrokerDebug.java

```diff
@@ -153,10 +153,15 @@ public Map<ServerInstance, List<String>> getRoutingTableForQuery(
       @Context HttpHeaders httpHeaders) {
     BrokerRequest brokerRequest = CalciteSqlCompiler.compileToBrokerRequest(query);
 
-    if (!_accessControlFactory.create()
-        .hasAccess(httpHeaders, TargetType.TABLE, brokerRequest.getQuerySource().getTableName(),
-            Actions.Table.GET_ROUTING)) {
-      throw new WebApplicationException("Permission denied", Response.Status.FORBIDDEN);
+    // TODO: Handle nested queries
+    if (brokerRequest.isSetPinotQuery() && brokerRequest.getPinotQuery().dataSource.isSetTableName()) {
```

Comment:
This can still cause NPE because `dataSource` might not be set in `PinotQuery`
```suggestion
    if (brokerRequest.isSetQuerySource() && brokerRequest.getQuerySource().isSetTableName()) {
```